pennylane/labs/dla/lie_closure_dense.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2024 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""A function to compute the Lie closure of a set of operators"""
+# pylint: disable=too-many-arguments
+
+import warnings
+from collections.abc import Iterable
+from typing import Union
+
+import numpy as np
+
+import pennylane as qml
+from pennylane.operation import Operator
+from pennylane.pauli import PauliSentence, PauliWord
+
+
+def _hermitian_basis(matrices: Iterable[np.ndarray], tol: float = None, subbasis_length: int = 0):
+    """Find a linearly independent basis of a list of (skew-) Hermitian matrices
+
+    Args:
+        matrices (Iterable[numpy.ndarray]): A list of Hermitian matrices.
```

Comment:
Could as well document this as a `numpy.ndarray` input, because it's a private method and we're always passing in an array (I think?). Then anyone reusing the function will stick to the established pattern of usage :)

Reply:
✅